### PR TITLE
kvdb/sqlbase: fix params used in `randRetryDelay`

### PR DESCRIPTION
**Pull request by yyforyongyu**
Realized the params used in `randRetryDelay`, causing us to always wait for 5s after the first attempt.

### CODE REVIEW

**Review by guggero — CHANGES_REQUESTED** [review]
Looks like the whole `sqlbase` package doesn't currently compile on master. The build isn't failing because we're still pinning a previous tag.

The main reason for this not being noticed is that we don't have any unit tests in that package. Otherwise we'd catch compilation errors of the _current_ code in submodules (even if we don't reference it in the main `go.mod`)

**Comment by yyforyongyu** [comment]
> The main reason for this not being noticed is that we don't have any unit tests in that package. Otherwise we'd catch compilation errors of the current code in submodules (even if we don't reference it in the main go.mod)

I don't think it's the case as there is a `TestInterface` in `kvdb/sqlite`. Without updating `go.mod` I cannot get it run. How about we use `replace github.com/lightningnetwork/lnd/kvdb  => ./kvdb` in our root `go.mod`?

**Comment by guggero** [comment]
> > The main reason for this not being noticed is that we don't have any unit tests in that package. Otherwise we'd catch compilation errors of the current code in submodules (even if we don't reference it in the main go.mod)


Then I think we need to change the way we run the unit tests. Or add the `kvdb_sqlite` build tag to the unit test tags?

> I don't think it's the case as there is a `TestInterface` in `kvdb/sqlite`. Without updating `go.mod` I cannot get it run. How about we use `replace github.com/lightningnetwork/lnd/kvdb => ./kvdb` in our root `go.mod`?

We used to have local replaces. But the problem is that those don't propagate when `lnd` is used as a library. Then the version in the `go.mod` is used. So we often forgot to bump that version and then ended up needing to fix the `go.mod` after already having released a version. This forces us to update the `go.mod` to even get a new functionality into `lnd` in the first place.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -217,6 +217,25 @@ jobs:
           path-to-profile: coverage.txt
           parallel: true
 
+  ###############################
+  # run unit tests for submodules
+  ###############################
+  submodule-unit-test:
```
Comment (guggero) [inline .github/workflows/main.yml:223]: Can we just add `- unit-module` to the matrix in the job above?

File: scripts/unit_test_modules.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+echo "testing kvdb with sqlite..."
+go test -tags="kvdb_sqlite" -timeout=5m || exit 1
+
+echo "testing kvdb with postgres..."
+go test -tags="kvdb_postgres" -timeout=5m || exit 1
+
+echo "testing kvdb with etcd..."
+go test -tags="kvdb_etcd" -timeout=5m || exit 1
+
+# TODO(yy): add all the submodules here.
```
Comment (guggero) [inline scripts/unit_test_modules.sh:20]: Nice idea, here's my suggestion for resolving the TODO:

```
#!/bin/bash

IGNORE="tools"
SUBMODULES=$(find . -mindepth 2 -name "go.mod" | cut -d'/' -f2 | grep -v "$IGNORE")

for submodule in $SUBMODULES
do
  pushd $submodule

  echo "Running submodule unit tests in $(pwd)"
  echo "testing $submodule..."
  go test -timeout=5m || exit 1
  
  if [[ "$submodule" == "kvdb" ]]
  then
      echo "testing $submodule with sqlite..."
      go test -tags="kvdb_sqlite" -timeout=5m || exit 1
      
      echo "testing $submodule with postgres..."
      go test -tags="kvdb_postgres" -timeout=5m || exit 1
      
      echo "testing $submodule with etcd..."
      go test -tags="kvdb_etcd" -timeout=5m || exit 1
  fi
  
  popd
done
```

File: Makefile [hunk 3]
```diff
@@ -184,6 +184,10 @@ unit: $(BTCD_BIN)
 	@$(call print, "Running unit tests.")
 	$(UNIT)
 
+unit-module:
+	@$(call print, "Running submodule unit tests.")
+	scripts/unit_test_modules.sh
```
Comment (guggero) [inline Makefile:189]: Maybe we'll even always want to run this, so add the call to this script to the already existing `unit` goal?

**Review by guggero — COMMENTED** [review]
Great fix, thanks a lot!

File: .github/workflows/main.yml (hunk 1 above)
Comment (yyforyongyu) [reply .github/workflows/main.yml]: Done!

File: scripts/unit_test_modules.sh (hunk 2 above)
Comment (yyforyongyu) [reply scripts/unit_test_modules.sh]: nice nice thank you!

File: Makefile (hunk 3 above)
Comment (yyforyongyu) [reply Makefile:189]: I kinda like the current `unit` goal tho, as I use `make unit log="stdlog trace" pkg=$pkg case=$case` a lot to check existing tests. If we wanna combine `unit-module` and `unit`, I think we'll need more flags to run a specific test case? That being said, I also wanna do `make unit-module pkg=$pkg case=$case` someday, guess it's something we can add in a follow-up PR.

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

**Comment by guggero** [comment]
Looks like there are more go module changes needed. We'll address them in https://github.com/lightningnetwork/lnd/pull/7944

**Comment by yyforyongyu** [comment]
Lasted push added a new command `make tidy-module` to run `go mod tidy` for all modules and a new action using `make tidy-module-check` to check modules are always updated.

**Review by positiveblue — APPROVED** [review]
Thanks for adding the go mod tidy automation :100:

**Pull request merged**